script.speedtester/default.py

```diff
@@ -0,0 +1,939 @@
+
+        startST.append('Testing from %(isp)s (%(ip)s)' % config['client'])
+        self.update_textbox(startST)
+
+        best = getBestServer(servers)
+
+        try:
+            startST.append('Selecting best server based on latency')
+            self.update_textbox(startST)
+        except:pass
+        try:
+            startST.append('Hosted by: %(sponsor)s' % best)
+            self.update_textbox(startST)
+        except:pass
+        try:
+            startST.append('Host Server: %(host)s' % best)
+            self.update_textbox(startST)
+        except:pass
+        try:
+            startST.append('Country: %(country)s' % best)
+            self.update_textbox(startST)
+        except:pass
+        try:
+            startST.append('City , State: %(name)s' % best)
+            self.update_textbox(startST)
+        except:pass
+        try:
+            km2mi = 0.62
+            km = '%(d)0.2f ' % best
+            Distance = float(km)
+            miles = Distance * km2mi
+            startST.append('Distance: %s mi' % miles)
+            self.update_textbox(startST)
+        except:pass
+        try:
+            startST.append('Ping: %(latency)s ms' % best)
+            self.update_textbox(startST)
+            self.ping_textbox.setLabel("%.0f" % float(best['latency']))
+        except:pass
+        self.imgCentertext.setImage(' ')
+        self.imgPing.setEnabled(False)
+        self.imgPing_glow.setEnabled(False)
+
+        sizes = [350, 500, 750, 1000, 1500, 2000, 2500, 3000, 3500, 4000]
+        urls = []
+        for size in sizes:
+            for i in range(0, 4):
+                urls.append('%s/random%sx%s.jpg' %
+                                    (os.path.dirname(best['url']), size, size))
+        self.imgGauge.setVisible(True)
+        xbmc.Monitor().waitForAbort(1)
+        self.configGauge(0)
+        self.imgGauge_arrow.setVisible(True)
+
+        startST.append('Testing download speed')
+        self.update_textbox(startST)
+        dlspeed = self.downloadSpeed(urls, simple)
+        startST.append('Download: %0.2f M%s/s' % ((dlspeed / 1000 / 1000) * units[1], units[0]))
+        self.update_textbox(startST)
+        self.dl_textbox.setLabel("%.2f" % float((dlspeed / 1000 / 1000) * units[1]))
+
+        sizesizes = [int(.25 * 1000 * 1000), int(.5 * 1000 * 1000)]
+        sizes = []
+        for size in sizesizes:
+            for i in range(0, 25):
+                sizes.append(size)
+
+        startST.append('Testing upload speed')
+        self.update_textbox(startST)
+        ulspeed = self.uploadSpeed(best['url'], sizes, simple)
+        startST.append('Upload: %0.2f M%s/s' % ((ulspeed / 1000 / 1000) * units[1], units[0]))
+        self.update_textbox(startST)
+        self.ul_textbox.setLabel("%.2f" % float((ulspeed / 1000 / 1000) * units[1]))
+        self.configGauge(0, (ulspeed / 1000 / 1000) * 8, time=3000)
+        xbmc.Monitor().waitForAbort(2)
+
+        if share:
```

Comment:
Same below

Reply:
Same what ?

Reply:
Same what?

Reply:
hardcoded non-localized values

Reply:
I don't see any below, only log messages.

Reply:
disregard, it is just query arguments for the http request